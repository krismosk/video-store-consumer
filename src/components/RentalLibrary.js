import React, { Component } from 'react';
import Movie from './Movie';
import axios from 'axios';

class RentalLibrary extends Component {
  constructor(props) {
    super(props);

    this.state = {
      movies: [],
    };
  }

  componentDidMount () {
    axios.get('http://localhost:3003/movies')
    .then((response) => {
      this.setState({
        movies: response.data
      });
    })
    .catch((error) => {
      // come back to handle errors better
      console.log(error.message)
    });
  }

  makeMovieList() {
    const movieList = this.state.movies.map((movie, i) => {
      return <Movie
        key={i}
        title={movie.title}
        overview={movie.overview}
        releaseDate={movie.release_date}
        inventory={movie.inventory}
        image={movie.image_url}
        externalId={movie.external_id}
        selectMovieCallback={movie.selectMovieCallback}
      />
    });
    return movieList
  }

  render() {
    return (
      <div>
        { this.makeMovieList }
      </div>
    )
  }

}

export default RentalLibrary;