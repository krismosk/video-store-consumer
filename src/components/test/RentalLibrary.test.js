import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RentalLibrary from '../RentalLibrary';

const rentalLibrary = <RentalLibrary
  selectMovie = {() => { }}
/>;

describe("Rental Library", () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render ( rentalLibrary );

    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
})