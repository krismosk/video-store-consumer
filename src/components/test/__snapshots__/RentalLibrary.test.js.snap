// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rental Library it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
